tmt/utils.py

```diff
@@ -3271,6 +3271,25 @@ def git_root(*, fmf_root: Path, logger: tmt.log.Logger) -> Optional[Path]:
         return None
 
 
+def git_add(*, path: Path, logger: tmt.log.Logger) -> Optional[str]:
+    """
+    Add path to the git repository.
+
+    :param path: path to add to the git repository.
+    :param logger: used for logging.
+    :returns: error message if the command failed, or ``None`` on success.
+    """
+    path = path.resolve()
+
+    try:
+        Command("git", "add", str(path)).run(cwd=path.parent, logger=logger)
+        return None
+
+    except RunError as e:
+        message = " ".join(e.stderr.splitlines()) if e.stderr else ""
+        return f"Failed to add path '{path}' to git: '{message}'"
```

Comment:
Why not raise an exception instead of returning its content as a return value? An exception would provide exactly the same value but in a more expectable manner.

Reply:
That would probably be better. It should be fixed in ab7d879.